packages/jsii-pacmak/lib/targets/go.ts

```diff
@@ -78,27 +86,34 @@ export class Golang extends Target {
     // local build directory for this module. if
     // we do, add a "replace" directive to point to it instead of download from
     // the network.
-    const visit = (pkg: RootPackage) => {
-      for (const dep of pkg.packageDependencies) {
-        for (const baseDir of dirs) {
-          const moduleDir = tryFindLocalModule(baseDir, dep);
-          if (moduleDir) {
-            replace[dep.goModuleName] = moduleDir;
-          }
-        }
-
-        // recurse to transitive deps ("replace" is only considered at the top level go.mod)
-        visit(dep);
-      }
+    const visit = async (pkg: RootPackage): Promise<void> => {
+      return Promise.all(
+        pkg.packageDependencies.map((dep) =>
+          Promise.all(
+            dirs.map((baseDir) =>
+              tryFindLocalModule(baseDir, dep).then((moduleDir) => {
+                if (moduleDir) {
+                  replace[dep.goModuleName] = moduleDir;
+                }
+              }),
+            ),
+          ).then(() =>
+            // recurse to transitive deps ("replace" is only considered at the top level go.mod)
+            visit(dep),
+          ),
+        ),
+      ).then(() => void undefined);
```

Comment:
Why not use await?

Reply:
[`no-await-in-loop`](https://eslint.org/docs/rules/no-await-in-loop) is why I transformed this block. Now I used this particular form because it does not require the outer function closure to be `async`, which itself reduces the cost of it.

This is a bit of a micro-optimisation (I got a reflex of writing those these way now... cuz in some cases it matters - not su much here though). I'm open to re-formulate here if you feel this isn't legible enough.